src/core/components/scroll-to-top.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from "react"
+
+const ScrollToTopButton = () => {
+  const [isVisible, setIsVisible] = useState(false)
+  const [hideTimeout, setHideTimeout] = useState(null) 
+
+  const styles = {
+    button: {
+      position: "fixed",
+      bottom: "20px",
+      right: "20px",
+      padding: "10px 20px",
+      fontSize: "18px",
+      backgroundColor: "rgba(125, 132, 146, 0.8)",
+      color: "#fff",
+      border: "2px solid rgba(125, 132, 146, 0.8)",
+      borderRadius: "5px",
+      cursor: "pointer",
+      boxShadow: "0px 4px 6px rgba(0, 0, 0, 0.1), 0 0 15px rgba(125, 132, 146, 0.6)",
+      transition: "opacity 0.5s ease, box-shadow 0.3s ease",
+      opacity: isVisible ? 1 : 0,
+      pointerEvents: isVisible ? "auto" : "none",
+    },
+    icon: {
+      width: "20px",
+      height: "20px",
+      fill: "#fff",
+    },
+  }
+
+  const toggleVisibility = () => {
+    if (window.scrollY > 300) {
+      setIsVisible(true)
+
+      if (hideTimeout) {
+        clearTimeout(hideTimeout)
+      }
+
+      const timeout = setTimeout(() => {
+        setIsVisible(false)
+      }, 3000)
+
+      setHideTimeout(timeout)
+    } else {
+      setIsVisible(false)
+    }
+  }
+
+  const scrollToTop = () => {
+    window.scrollTo({
+      top: 0,
+      behavior: "smooth",
+    })
+  }
+
+  useEffect(() => {
```

Comment:
You’re adding an event listener to window inside `useEffect,` which is great, but since `toggleVisibility` depends on nothing external, it could be defined directly inside `useEffect` for clarity and potential memory efficiency.